### ISSUE TITLE
{REQ} Stop overwriting RealSettings on update.

### ISSUE THREAD

**Issue by KAmaia** [NathanKell/ModularFuelSystem#195]
Rather than over write RealSettings on update, is could we please not?

**Comment by blowfishpro** [comment]
I've never tried it myself, but it looks like it should be possible to overwrite values using ModuleManager, i.e. `@ RFSETTINGS:FINAL { ... }` in another file.  `RFSETTINGS` is definitely loaded from the game database (as opposed to directly from `RealSettings.cfg`.

The messy part is that the settings are initialized when the first thing requires them, and I'm only mostly sure that nothing is going to ask for the settings before MM runs.  If this is not the case though, feel free to tell me.  To check for sure, you should see`*RF* Loading RFSETTINGS global settings` in the log when it's first loaded

**Comment by Starwaster** [comment]
yes, it's possible to patch RFSETTINGS with Module Manager. It's just another config node with fields and values. The question is, what is the REAL problem? What is really being asked her and why?

**Comment by KAmaia** [comment]
I routinely lose my settings when an update is patched in and have to
manually edit them when I notice.  I'm asking that settings not be
overwritten on update from ckan

**Comment by NathanKell** [comment]
Make your settings an mm patch rather than editing the file.

**Comment by jwvanderbeck** [comment]
Hi KAmaia,

When a mod is distributed it includes **all** its files, including settings.  This is because in KSP Mods there is no concept of Update versus New Install.  It is all the same.  The reason your custom settings are replaced is because you are literally replacing your custom settings file with the one in the mod each time you update.  There is no way around this using CKAN.  If you manually install the mod, you could of course not copy this one file over, then you would retain your settings.  However by doing that you risk problems if the settings file changes for a good reason.

The only clean way around this is as @NathanKell says.  RF uses the standard MM config system, which means that you can put your custom settings into your own MM patch and have that overwrite the default settings with yours.  Then any time the base settings change, yours will still apply on top.

**Comment by blowfishpro** [comment]
@KAmaia just following up here to make sure everything discussed makes sense - feel free to ask if you have any more questions about the suggested solurion

**Comment by KAmaia** [comment]
@Joseph Wong - Not sure how to module manager a raw config file.

**Comment by Starwaster** [comment]
Put this in a config file, edit to suit your needs. If you need to alter the other settings then it will get a little more complex.  Tip: Put any personal config patches in a folder such as MyTweaks. You can migrate that folder to any new KSP installations so that your personal mod changes follow you around.

`@RFSETTINGS
{
	@engineMassMultiplier = 4.0
	@useRealisticMass = true
	@heatMultiplier = 1.0
	@debugBoilOff = True
	@globalConductionCompensation = False
	@ferociousBoilOff = False
}
`

**Comment by jwvanderbeck** [comment]
Shouldn't that have a pass like FINAL to ensure it runs after?

**Comment by Starwaster** [comment]
ONLY if there's a chance something else might try to _patch_ that node and you want to be sure your _patch_ runs after all other _patches_.

Otherwise, the sequence is this: KSP loads RFSETTINGS along with all other ConfigNodes. It processes everything that it can process, excluding all MM patches which fail because it doesn't recognize the syntax.

MM (and other plugins run) and MM executes all ConfigNodes that have valid MM syntax. I didn't think it likely that anything else might try to patch RFSETTINGS because I haven't heard of any mods that do so. If a :FINAL pass is desired then it certainly won't hurt anything to add that in but it's probably not necessary.

I did make some changes to the patch I supplied though which is to change @ to % on the various settings though it's probably not really necessary since most of them are actually already in the config.

And thank you for using 'ensure' rather than 'insure'.
